official/wide_deep/wide_deep.py

```diff
@@ -254,25 +254,26 @@ def eval_input_fn():
       tensors_to_log={'average_loss': loss_prefix + 'head/truediv',
                       'loss': loss_prefix + 'head/weighted_loss/Sum'})
 
-  # Train and evaluate the model every `flags.epochs_between_evals` epochs.
-  for n in range(flags_obj.train_epochs // flags_obj.epochs_between_evals):
-    model.train(input_fn=train_input_fn, hooks=train_hooks)
-    results = model.evaluate(input_fn=eval_input_fn)
-
-    # Display evaluation metrics
-    tf.logging.info('Results at epoch %d / %d',
-                    (n + 1) * flags_obj.epochs_between_evals,
-                    flags_obj.train_epochs)
-    tf.logging.info('-' * 60)
-
-    for key in sorted(results):
-      tf.logging.info('%s: %s' % (key, results[key]))
-
-    benchmark_logger.log_evaluation_result(results)
-
-    if model_helpers.past_stop_threshold(
-        flags_obj.stop_threshold, results['accuracy']):
-      break
+  with logger.benchmark_context(benchmark_logger):
```

Comment:
I'm struggling with this. On the one hand, it makes sense; on the other, we are adding more and more boilerplate and infrastructure into what is supposed to be TF model code just for the sake of getting streaming benchmarks out. Is there a way that is less intrusive? An official.main() that wraps some of the benchmarking code? I am open to discussion here. I just fear we are crossing the line into making ourselves less usable for our primary users.